tests/cpu-vm

```diff
@@ -79,16 +73,40 @@ for i in $(seq "${cpuCount}" -1 1); do
   [ "$(lxc exec v1 -- ls /sys/devices/system/cpu | grep -xEc 'cpu[[:digit:]]+')" -eq "${i}" ]
 done
 
-# Try doing pinning while VM is running (shouldn't work)
-! lxc config set v1 limits.cpu=1,2 || false
+echo "==> Check that there is no CPU pinning set"
+QEMU_PID=$(lxc info v1 | awk '/^PID:/ {print $2}')
+! taskset --cpu-list -a -p "${QEMU_PID}" | grep -E ':\s+[0-9]+$' || false
+taskset --cpu-list -a -p "${QEMU_PID}" | grep "0-$((cpuCount-1))"
 
 # Set max CPU count
 lxc config set v1 limits.cpu="${cpuCount}"
 [ "$(lxc exec v1 -- ls /sys/devices/system/cpu | grep -xEc 'cpu[[:digit:]]+')" -eq "${cpuCount}" ]
 
-# check that CPU affinity is automatically set if feature present
-QEMU_PID=$(lxc info v1 | awk '/^PID:/ {print $2}')
-if journalctl --quiet --no-hostname --no-pager --boot=0 --unit=snap.lxd.daemon.service | grep "Scheduler: virtual-machine"; then
+# Unset CPU limit
+lxc config unset v1 limits.cpu
+
+# Unsetting the limit should leave the VM with 1 CPU
+[ "$(lxc exec v1 -- ls /sys/devices/system/cpu | grep -xEc 'cpu[[:digit:]]+')" -eq "1" ]
+
+if hasNeededAPIExtension limits_cpu_pin_strategy; then
+  echo "==> Check CPU auto pinning when limits.cpu.pin_strategy=auto"
+
+  # Stop VM
+  lxc stop -f v1
+  lxc launch "${IMAGE}" v1 --vm -c limits.cpu.pin_strategy=auto -s "${poolName}" --ephemeral
+  waitInstanceReady v1
+  lxc info v1
+
+  lxc config set v1 limits.cpu="${cpuCount}"
+  [ "$(lxc exec v1 -- ls /sys/devices/system/cpu | grep -xEc 'cpu[[:digit:]]+')" -eq "${cpuCount}" ]
+
+  # Try doing pinning while VM is running (shouldn't work)
+  ! lxc config set v1 limits.cpu=1,2 || false
```

Comment:
I'd move that back out of the `if` as this doesn't depend on the API's presence.